controllers/cloudflareCache.js

```diff
@@ -0,0 +1,78 @@
+const logsQuery = require("../models/logs");
+const cloudflare = require("../services/cloudflareService");
+const { logType } = require("../constants/logs");
+const { MAX_CACHE_PURGE_COUNT } = require("../constants/cloudflareCache");
+const { SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
+
+/**
+ * Purges the Cache of Members Profile Page
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const purgeMembersCache = async (req, res) => {
+  try {
+    const { id, username } = req.userData;
+    const logs = await logsQuery.fetchCacheLogs(id);
+    const logsCount = logs.length;
+
+    const files = [`https://members.realdevsquad.com/${username}`];
+
+    if (logsCount < MAX_CACHE_PURGE_COUNT) {
```

Comment:
What if a superuser is purging the cache?

Reply:
Explained the reason above as users should not be reached to superuser everytime they need the updated members page of theirs.

Reply:
But the superuser must be able to purge the cache as many times as they want.

Discussed it today with ankush in morning

Reply:
So I think for superuser there should be a separate API to purge the cache and separate UI on dashboard site as `username` will be expected in the body to purge the cache of any user. Or I think we can modify this API to adapt the behaviour that will make this API a little bit complex probably

Reply:
Implemented in the same controller function in my local and it will look like this
![image](https://user-images.githubusercontent.com/78433013/210129423-8f856e0b-0beb-444d-95f3-3162f7bfd85d.png)

Reply:
Not the best code as there a lot of repetition. I can simplify this by making a separate controller function which just clears the cache and put the restriction in a new controller function which will be eventually called by route

Reply:
Simplified a bit
![image](https://user-images.githubusercontent.com/78433013/210129893-9ca74e52-908b-41b6-9b36-d623acc43b6a.png)

Reply:
cant you identify the super user from the req.userData?
then in the same controller add a check if it's a normal user then use max count if the superuser then don't use max count.

Reply:
That is what I have implemented actually, see the above screenshot

Reply:
Added the changes